### ISSUE TITLE
7zip version used

### ISSUE THREAD

**Issue by jerkstore369** [aonez/Keka#400]
I noticed Keka uses 7zip v16.02, or rather p7zip based on 7zip 16.02 I guess. It seems severely outdated. It doesn't seem like p7zip has been updated in a while. I think myspace has disappeared and isn't working on it anymore. I can't believe there's nobody working on it. Isn't that an issue? Like, aren't there many new features and bugfixes that 7zip users on macOS are missing out on?

**Comment by aonez** [comment]
Yep, using p7zip 16.02 with all available security and fix patches. Would be awesome to have an updated version of p7zip but someone has to do the work. Maybe starting a repository would be ideal.

**Comment by wesinator** [comment]
p7zip 16.02 is the stable version offered by brew, and seems to be the latest version on Unix.

**Comment by jerkstore369** [comment]
Yes, I get that wesinator, I just find it hard to believe nobody is working on an updated version. I'm not a programmer so I wouldn't know the first thing.

**Comment by sikhness** [comment]
Can the new LZMA SDK not be used to implement the latest version of 7z in Keka?

**Comment by aonez** [comment]
@sikhness `p7zip` is used for various formats, not only the LZMA algorithm.

**Comment by sikhness** [comment]
@aonez thanks for clarification!  Is it possible to include the LZMA SDK then just for LZMA compression only instead of using p7zip for that?  This way we can use the latest version for LZMA compression for 7zip formats?

**Comment by aonez** [comment]
@sikhness sure! I'll investigate this option 👍🏼

**Comment by Khoulaiz** [comment]
I got 7zip files from the windows world (encrypted with a password), that I couldn't uncompress with Keka. I had to use unarchiver to do it. I cannot share the file, it contains sensitve information but it looks like that an update of the 7zip engine is important.

**Comment by aonez** [comment]
@Khoulaiz can you share more info, possibly a [log](https://github.com/aonez/Keka/wiki/Report-an-issue) and the password used? You can also send those details at info@keka.io, if you prefer.
